src/tinypeel/Peeling/PeelingIO.py

```diff
@@ -118,6 +119,30 @@ def writeGenoProbs(pedigree, genoProbFunc, outputFile):
                 )
 
 
+def writeGenoProbs(pedigree, genoProbFunc, outputFile):
+    with open(outputFile, "w+") as f:
+        for idx, ind in pedigree.writeOrder():
+            matrix = genoProbFunc(ind.idn)
+            f.write("\n")
+            for i in range(matrix.shape[0]):
```

Comment:
@AprilYUZhang My Python is at an entry-level, but I think this loop is wrong. We will loop from 0 to 4 and we effectively want to sum up rows 0&1 and 2&3. At the moment you are summing up 1+2 (instead of 0+1) and you are only outputting 2 in the elif part (instead of 2+3). Does this make sense?

Reply:
@gregorgorjanc What you interpret from the code is absolutely correct, but for the matrix of genotype probability, each individual has 4 lines to represent the probabilities of having aa, aA, Aa, AA alleles respectively. So, if we want the unphased genotype probability, we need to add up the second and the third row for each individual.

Reply:
@XingerTang yeah, you are right, so we should output 0, 1+2, and 3, which is what the code does! Doh. Sorry for the noise @AprilYUZhang ;)

To avoid such blunders, it would help to add some comments to the code. What about this:

```
def writeGenoProbs(pedigree, genoProbFunc, outputFile):
     with open(outputFile, "w+") as f:
         for idx, ind in pedigree.writeOrder():
             matrix = genoProbFunc(ind.idn)
             f.write("\n")
             for i in range(matrix.shape[0]):
                if i == 1: # Add up probabilities for aA and Aa
                     f.write(
                         ind.idx
                         + " "
                         + " ".join(
                             map("{:.4f}".format, matrix[i, :] + matrix[i + 1, :])
                         )
                         + "\n"
                     )
                 elif i != 2: # Print probabilities for aa and AA
                     f.write(
                         ind.idx
                         + " "
                         + " ".join(map("{:.4f}".format, matrix[i, :]))
                         + "\n"
                     )
```

Reply:
@gregorgorjanc I think the comments are very clear.

Reply:
@AprilYUZhang don't forget to add the comments to the code